modules/Core/src/main/java/org/terasology/rendering/nui/layers/ingame/inventory/InventoryGrid.java

```diff
@@ -55,6 +55,23 @@
     private List<InventoryCell> cells = Lists.newArrayList();
     private Binding<EntityRef> targetEntity = new DefaultBinding<>(EntityRef.NULL);
 
+    private InteractionListener interactionListener = new BaseInteractionListener() {
+        @Override
+        public boolean onMouseClick(NUIMouseClickEvent event) {
+            MouseInput mouseButton = event.getMouseButton();
+            if (mouseButton == MouseInput.MOUSE_LEFT) {
+                LocalPlayer localPlayer = CoreRegistry.get(LocalPlayer.class);
+                EntityRef movingItem = CoreRegistry.get(LocalPlayer.class).getCharacterEntity().getComponent(CharacterComponent.class).movingItem;
```

Comment:
oh and new code should avoid core registry. you can get a instance of LocalPlayer by defining a field annotated with `@In` in UI screens and systems.